manifest-build-tools/application/release_debian_packages.py

```diff
@@ -0,0 +1,191 @@
+
+        if self._component:
+            cmd_args += ["--component", self._component]
+        if self._distribution:
+            cmd_args += ["--distribution", self._distribution]
+        if self._architecture:
+            cmd_args += ["--architecture", self._architecture]
+
+        cmd_args += ["--package", package]
+        cmd_args += ["--version", version]
+        cmd_args += ["--file_path", file_path]
+
+        try:
+            common.run_command(cmd_args)
+        except Exception, ex:
+            raise RuntimeError("Failed to upload file {0} due to {1}".format(file_path, ex))
+        return True
+
+def parse_args(args):
+    """
+    Parse script arguments.
+    :return: Parsed args for assignment
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--build-directory',
+                        required=True,
+                        help="Top level directory that stores all the cloned repositories.",
+                        action='store')
+
+    parser.add_argument('--debian-depth',
+                        help="The depth in top level directory that you want"
+                             " this program look into to find debians.",
+                        default=3,
+                        type=int,
+                        action='store')
+
+    parser.add_argument('--bintray-credential',
+                        required=True,
+                        help="bintray credential for CI services: <Credentials>",
+                        action='store')
+
+    parser.add_argument('--bintray-subject',
+                        required=True,
+                        help="the Bintray subject, which is either a user or an organization",
+                        action='store')
+
+    parser.add_argument('--bintray-repo',
+                        required=True,
+                        help="the Bintary repository name",
+                        action='store')
+
+    parser.add_argument('--bintray-component',
+                        help="such as: main",
+                        action='store',
+                        default='main')
+
+    parser.add_argument('--bintray-distribution',
+                        help="such as: trusty, xenial",
+                        action='store',
+                        default='trusty')
+
+    parser.add_argument('--bintray-architecture',
+                        help="such as: amd64, i386",
+                        action='store',
+                        default='amd64')
+
+    parsed_args = parser.parse_args(args)
+    return parsed_args
+
+def get_push_executable():
+    repo_dir = os.path.dirname(sys.path[0])
+    for subdir, dirs, files in os.walk(repo_dir):
+        for file in files:
+            if file == push_exe_script:
+                return os.path.join(subdir, file)
+    return None
+
+def upload_debs(build_directory, debian_depth, bintray):
+    """
+    Upload the debian packages under top level directory.
```

Comment:
How about make it more verbose ?

example: it will walk through all sub-folder under $build_directory, and for every *.deb found:
1. retrieve its version
2. upload to bintray with this version